aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/usecases/ConfirmSignUpUseCase.kt

```diff
@@ -0,0 +1,51 @@
+package com.amplifyframework.auth.cognito.usecases
+
+import com.amplifyframework.auth.AuthException
+import com.amplifyframework.auth.cognito.AuthEnvironment
+import com.amplifyframework.auth.cognito.CognitoAuthExceptionConverter
+import com.amplifyframework.auth.cognito.helpers.SRPHelper
+import com.amplifyframework.auth.options.AuthConfirmSignUpOptions
+import com.amplifyframework.auth.result.AuthSignUpResult
+import com.amplifyframework.auth.result.step.AuthNextSignUpStep
+import com.amplifyframework.auth.result.step.AuthSignUpStep
+import com.amplifyframework.core.Consumer
+
+internal class ConfirmSignUpUseCase(environment: AuthEnvironment) {
+    private val logger = environment.logger
+    private val configuration = environment.configuration
+    private val client = environment.cognitoAuthService.cognitoIdentityProviderClient
+
+    suspend fun execute(
+        username: String,
+        confirmationCode: String,
+        options: AuthConfirmSignUpOptions,
+        onSuccess: Consumer<AuthSignUpResult>,
+        onError: Consumer<AuthException>
+    ) {
+        logger?.verbose("ConfirmSignUp Starting execution")
+        try {
+            client?.confirmSignUp {
+                this.username = username
+                this.confirmationCode = confirmationCode
+                this.clientId = configuration.userPool?.appClient
+                this.secretHash = SRPHelper.getSecretHash(
+                    username,
+                    configuration.userPool?.appClient,
+                    configuration.userPool?.appClientSecret
+                )
+            }
+
+            val authSignUpResult = AuthSignUpResult(
+                true,
+                AuthNextSignUpStep(AuthSignUpStep.DONE, mapOf(), null),
+                null
+            )
+            onSuccess.accept(authSignUpResult)
```

Comment:
This and line 27 can execute concurrently can't they?

Reply:
No, we would want to wait for the service call to either be successful or throw an error in which case we will call error callback.